src/Abstractions/NexusMods.Abstractions.Jobs/AJobWorker.cs

```diff
@@ -1,69 +1,181 @@
 
-    public Task StartAsync(CancellationToken cancellationToken = default)
+    private async Task<OneOf<JobResult, Paused>> ExecuteAsyncInner(AJob job, CancellationToken cancellationToken)
     {
-        
+        try
+        {
+            var result = await ExecuteAsync(job, cancellationToken);
+            return OneOf<JobResult, Paused>.FromT0(result);
+        }
+        catch (TaskCanceledException)
+        {
+            if (job.IsRequestingPause)
+            {
+                job.IsRequestingPause = false;
+                return new Paused();
+            }
 
-        throw new NotImplementedException();
+            return JobResult.CreateCancelled();
+        }
+        catch (Exception e)
+        {
+            return JobResult.CreateFailed(e);
+        }
     }
 
-    public Task PauseAsync(CancellationToken cancellationToken = default)
+    private void SetWorker(AJob job)
     {
-        throw new NotImplementedException();
+        if (job.Worker is null)
+        {
+            job.SetWorker(this);
+            return;
+        }
+
+        Debug.Assert(ReferenceEquals(job.Worker, this), "same worker");
     }
 
-    public Task CancelAsync(CancellationToken cancellationToken = default)
+    /// <inheritdoc/>
+    public ValueTask StartAsync(IJob input, CancellationToken cancellationToken = default)
     {
-        throw new NotImplementedException();
+        if (input is not AJob job) throw new NotSupportedException();
+        cancellationToken.ThrowIfCancellationRequested();
+
+        SetWorker(job);
+        if (job.Status == JobStatus.None)
+        {
+            job.SetStatus(JobStatus.Created);
+        }
+
+        if (job.Status == JobStatus.Paused)
+        {
+            Debug.Assert(job.Task is null || job.Task.IsCompleted, "task should've completed");
+            job.Task = null;
+            var didReset = job.CancellationTokenSource.TryReset();
+            Debug.Assert(didReset, "reset should work");
+        }
+
+        job.SetStatus(JobStatus.Running);
+
+        Task<OneOf<JobResult, Paused>> task;
+        try
+        {
+            task = ExecuteAsyncInner(job, job.CancellationTokenSource.Token);
+        }
+        catch (Exception e)
+        {
+            // synchronous or async startup exception
+            job.SetResult(JobResult.CreateFailed(e), inferStatus: true);
+            return ValueTask.CompletedTask;
+        }
+
+        // task might complete synchronously
+        if (task.IsCompleted)
+        {
+            Debug.Assert(task.IsCompletedSuccessfully, "wrapper task should always complete successfully");
+            var result = task.Result;
+            if (result.IsT0)
+            {
+                job.SetResult(result.AsT0, inferStatus: true);
+            }
+            else
+            {
+                job.SetStatus(JobStatus.Paused);
+            }
+
+            job.Task = null;
+            return ValueTask.CompletedTask;
+        }
+
+        task.Start();
```

Comment:
From what I understand `Task.Start()` should be problematic here, since the task would count as already being in a started state

Reply:
Why would it already be in the started state?

Reply:
For this I think:
`task = ExecuteAsyncInner(job, job.CancellationTokenSource.Token);`
the task should already start executing until the first await is encountered.
Calling start again should cause invalidoperation exception from the docs

Reply:
True, fixed.